src/lesson4/LinkIterator.java

```diff
@@ -0,0 +1,78 @@
+package lesson4;
+
+public class LinkIterator<E> {
+    private LinkedList.Node<E> current;
+    private LinkedList.Node<E> previous;
+    private SimpleLinkedListImpl<E> collection;
+
+    public LinkIterator(SimpleLinkedListImpl<E> collection) {
+        this.collection = collection;
+        this.reset();
+    }
+
+    public void reset(){
+        current = collection.firstElement;
+        previous = null;
+    }
+
+    public boolean atEnd(){
+        return current.next == null;
```

Comment:
```suggestion
        return current== null;
```
иначе не работает для случая коллекции с одним элементом